hudi-client/src/main/java/org/apache/hudi/table/action/rollback/CopyOnWriteRollbackActionExecutor.java

```diff
@@ -80,10 +82,16 @@ public CopyOnWriteRollbackActionExecutor(JavaSparkContext jsc,
     if (!resolvedInstant.isRequested()) {
       // delete all the data files for this commit
       LOG.info("Clean out all base files generated for commit: " + resolvedInstant);
-      stats = getRollbackStrategy().execute(resolvedInstant);
+      if(doDelete) {
+        stats = getRollbackStrategy().execute(resolvedInstant);
+      } else{
+        stats = executeRollbackUsingFileListing(resolvedInstant, false);
```

Comment:
and if doDelete is false, we call into executeRollbackUsingFileListing directly.